internal/accounts/create-interactive.go

```diff
@@ -80,145 +102,161 @@ func createInteractive(state *flowkit.State) error {
 	}
-	log.Info(fmt.Sprintf(
-		"%s New account created with address %s and name %s on %s network.\n",
-		output.SuccessEmoji(),
-		output.Bold(fmt.Sprintf("0x%s", account.Address().String())),
-		output.Bold(name),
-		output.Bold(networkName)),
-	)
-
 	outputList(log, items, false)
 
 	return nil
 }
 
-var (
-	testAddress = ""
-	mainAddress = ""
-	testKey     = ""
-	mainKey     = ""
-)
-
-// createAccount on the network using the available signers
-func createAccount(
+// createNetworkAccount using the account creation API and return the newly created account address.
+func createNetworkAccount(
+	state *flowkit.State,
+	services *services.Services,
 	name string,
 	key crypto.PrivateKey,
+	privateFile string,
 	network config.Network,
-	state *flowkit.State,
-	service *services.Services,
 ) (*flowkit.Account, error) {
-	privateFile := fmt.Sprintf("%s.pkey", name)
+	networkAccount := &lilicoAccount{
+		PublicKey: strings.TrimPrefix(key.PublicKey().String(), "0x"),
+	}
 
-	var (
-		accountKey flowkit.AccountKey
-		signer     *flowkit.Account
-		err        error
-	)
+	id, err := networkAccount.create(network.Name)
+	if err != nil {
+		return nil, err
+	}
 
-	if network == config.DefaultEmulatorNetwork() {
-		signer, err = state.EmulatorServiceAccount()
-		if err != nil {
-			return nil, err
-		}
-		accountKey = flowkit.NewHexAccountKeyFromPrivateKey(0, crypto.SHA3_256, key)
+	result, err := getAccountCreationResult(services, id)
+	if err != nil {
+		return nil, err
+	}
 
-	} else {
-		signer = createSigner(network)
-		accountKey = flowkit.NewFileAccountKey(privateFile, 0, crypto.ECDSA_P256, crypto.SHA3_256)
-		err = util.AddToGitIgnore(privateFile, state.ReaderWriter())
-		if err != nil {
-			return nil, err
-		}
-		// create the private key file
-		err = state.ReaderWriter().WriteFile(privateFile, []byte(key.String()), os.FileMode(0644))
-		if err != nil {
-			return nil, fmt.Errorf("failed saving private key: %w", err)
-		}
+	events := flowkit.EventsFromTransaction(result)
+	address := events.GetCreatedAddresses()
+	if len(address) == 0 {
+		return nil, fmt.Errorf("account creation error")
 	}
+
+	err = util.AddToGitIgnore(privateFile, state.ReaderWriter())
 	if err != nil {
 		return nil, err
 	}
 
-	address, err := sendCreateAccountTransaction(signer, key, network, service)
+	err = state.ReaderWriter().WriteFile(privateFile, []byte(key.String()), os.FileMode(0644))
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed saving private key: %w", err)
 	}
 
-	return flowkit.NewAccount(name).SetAddress(*address).SetKey(accountKey), nil
+	return flowkit.NewAccount(name).SetAddress(*address[0]).SetKey(
+		flowkit.NewFileAccountKey(privateFile, 0, crypto.ECDSA_P256, crypto.SHA3_256),
```

Comment:
Might be nice to define the crypto as constants since it's used everywhere
```
const defaultSignAlg = crypto.ECDSA_P256
const defaultHashAlg = crypto.SHA3_256
```

Reply:
@DylanTinianov addressed